src/controllers/expenses.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const userService = require('../services/users');
+const expenseService = require('../services/expenses');
+
+const getAll = (req, res) => {
+  const filters = req.query;
+  const expenses = expenseService.getAll(filters);
+
+  res.send(expenses);
+};
+
+const getOne = (req, res) => {
+  const { expenseId } = req.params;
+
+  const foundExpense = expenseService.findById(expenseId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(foundExpense);
+};
+
+const add = (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  const isDataValid = typeof userId !== 'number'
+    || typeof spentAt !== 'string'
+    || typeof title !== 'string'
+    || typeof amount !== 'number'
+    || typeof category !== 'string'
+    || typeof note !== 'string';
+
```

Comment:
maybe you should change you condition, because isDataValid it’s true, but your condition means not valid date

Reply:
Agree

Reply:
fixed